### PR TITLE
Add verify-enterprise-contract-v2 task

### PR DESCRIPTION
**Pull request by lcarva**
https://issues.redhat.com/browse/HACBS-535

Signed-off-by: Luiz Carvalho <lucarval@redhat.com>

### CODE REVIEW

**Comment by lcarva** [comment]
@redhat-appstudio/enterprise-contract ptal

File: hack/start-verify-ec-task-v2.sh [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+#      ./start-verify-ec-task-v2.sh <image-ref>
+#    where <image-ref> is a valid image reference, e.g. quay.io/spam/bacon@sha256:...
+#
+set -euo pipefail
+
+IMAGE_REF="$1"
+
+PUBLIC_KEY_SECRET="${PUBLIC_KEY_SECRET:-cosign-public-key}"
+
+NAMESPACE="$(oc get sa default -o jsonpath='{.metadata.namespace}')"
```
Comment (zregvart) [inline hack/start-verify-ec-task-v2.sh:32]: Nit: bit straighter way of getting current namespace

```suggestion
NAMESPACE="$(oc project -q)"
```

File: pipelines/hacbs/e2e-ec.yaml [hunk 2]
```diff
@@ -81,3 +88,21 @@ spec:
...
+    - name: verify-enterprise-contract-v2
+      params:
+        - name: PUBLIC_KEY
+          value: "$(params.PUBLIC_KEY)"
+        - name: POLICY_CONFIGURATION
+          value: "$(params.POLICY_CONFIGURATION)"
+        - name: IMAGES
+          value: "$(params.IMAGES)"
+        - name: COSIGN_EXPERIMENTAL
+          value: "$(params.COSIGN_EXPERIMENTAL)"
```
Comment (zregvart) [inline pipelines/hacbs/e2e-ec.yaml:100]: Do we need this?

File: tasks/verify-enterprise-contract-v2.yaml [hunk 3]
```diff
@@ -0,0 +1,102 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: verify-enterprise-contract-v2
+  annotations:
+    tekton.dev/displayName: Verify Enterprise Contract V2
+    tekton.dev/pipelines.minVersion: "0.19"
+    tekton.dev/tags: cosign, chains, signature, conftest
```
Comment (zregvart) [inline tasks/verify-enterprise-contract-v2.yaml:9]:
```suggestion
    tekton.dev/tags: ec, chains, signature, conftest
```

File: tasks/verify-enterprise-contract-v2.yaml [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+      description: >-
+        Public key used to verify signatures. Must be a valid k8s cosign
+        reference, e.g. k8s://my-space/my-secret where my-secret contains
+        the expected cosign.pub attribute.
+      default: "k8s://$(context.taskRun.namespace)/cosign-public-key"
+
+    - name: COSIGN_EXPERIMENTAL
+      type: string
+      description: Control transparency log lookups. Set to "0" to disable it.
+      default: "1"
```
Comment (zregvart) [inline tasks/verify-enterprise-contract-v2.yaml:41]: I don't think we need this

File: tasks/verify-enterprise-contract-v2.yaml [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+          # The Tekton Operator automatically sets the SSL_CERT_DIR env to the value below but,
+          # of course, without the $(param.SSL_CERT_DIR) bit. When a Task Step sets it to a
+          # value, the Tekton Operator does not do any processing of the value. However, Tekton
+          # Pipelines will fail to execute because some of these values are required for its
+          # execution. As a workaround, append the SSL_CERT_DIR value from params to the default
+          # value expected by Tekton Pipelines. NOTE: If params.SSL_CERT_DIR is empty, the value
+          # will contain a trailing ":" - this is ok.
+          value: "/tekton-custom-certs:/etc/ssl/certs:/etc/pki/tls/certs:/system/etc/security/cacerts:$(params.SSL_CERT_DIR)"
+        - name: COSIGN_EXPERIMENTAL
+          value: $(params.COSIGN_EXPERIMENTAL)
```
Comment (zregvart) [inline tasks/verify-enterprise-contract-v2.yaml:102]: I don't think we need this

**Review by zregvart — COMMENTED** [review]
+1 looks good, few nits in the comments

File: hack/start-verify-ec-task-v2.sh (hunk 1 above)
Comment (lcarva) [reply hack/start-verify-ec-task-v2.sh]: Nice!

**Review by zregvart — APPROVED** [review]

**Pull request merged**